### DISCUSSION TITLE
linux demo 渲染alpha.pag得到的bmp图片背景是黑的

### DISCUSSION THREAD

**Discussion by rebornfz** [Tencent/libpag#360] [Q&A] [answered]
linux demo 渲染alpha.pag得到的bmp图片背景是黑的
pagplayer里查看是透明的背景：
![image](https://user-images.githubusercontent.com/27042311/172336681-a41205d4-f20f-4818-9759-37ed3f19d7f6.png)
生成的bmp图片的背景是黑的：
![image](https://user-images.githubusercontent.com/27042311/172337682-cca75c14-bd1a-417a-bd59-7349df4b35f9.png)

**Comment by kevingpqi123** [comment, marked as answer]
这里透明色默认选择的是黑色填充，demo 工程提供了 Linux 如何获取 PAG 渲染数据，在实际业务使用场景中，一般是将渲染数据 （RGBA 或 BGRA 格式）转换成 YUV 数据进行视频编码，或者借助于图片编码库（如png、webp等）编码导出成图片。
